diff -rupd boost_1_70_0_orig/tools/boost_install/boost-install.jam boost_1_70_0/tools/boost_install/boost-install.jam
--- boost_1_70_0_orig/tools/boost_install/boost-install.jam	2019-04-09 21:36:54.000000000 +0200
+++ boost_1_70_0/tools/boost_install/boost-install.jam	2019-06-17 16:06:11.466579439 +0200
@@ -156,7 +156,7 @@ rule generate-cmake-variant- ( target :
             "  return()"
             "endif()"
             ""
-            "if(BUILD_SHARED_LIBS)"
+            "if(BUILD_SHARED_LIBS AND NOT Boost_USE_STATIC_LIBS)"
             "  _BOOST_SKIPPED(\"$(fname)\" \"static, BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}, set Boost_USE_STATIC_LIBS=ON to override\")"
             "  return()"
             "endif()"
