diff -rupdN libharu-2.3.0RC2.orig/include/hpdf_config.h.cmake libharu-2.3.0RC2/include/hpdf_config.h.cmake
--- libharu-2.3.0RC2.orig/include/hpdf_config.h.cmake	Thu Jan  1 01:00:00 1970
+++ libharu-2.3.0RC2/include/hpdf_config.h.cmake	Sun Jan  9 23:22:14 2011
@@ -0,0 +1,70 @@
+/* include/hpdf_config.h.in.  Generated from configure.in by autoheader.  */
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#cmakedefine LIBHPDF_HAVE_DLFCN_H
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#cmakedefine LIBHPDF_HAVE_INTTYPES_H
+
+/* Define to 1 if you have the `png' library (-lpng). */
+#cmakedefine LIBHPDF_HAVE_LIBPNG
+
+/* Define to 1 if you have the `z' library (-lz). */
+#cmakedefine LIBHPDF_HAVE_LIBZ
+
+/* Define to 1 if you have the <memory.h> header file. */
+#cmakedefine LIBHPDF_HAVE_MEMORY_H
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#cmakedefine LIBHPDF_HAVE_STDINT_H
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#cmakedefine LIBHPDF_HAVE_STDLIB_H
+
+/* Define to 1 if you have the <strings.h> header file. */
+#cmakedefine LIBHPDF_HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#cmakedefine LIBHPDF_HAVE_STRING_H
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#cmakedefine LIBHPDF_HAVE_SYS_STAT_H
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#cmakedefine LIBHPDF_HAVE_SYS_TYPES_H
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#cmakedefine LIBHPDF_HAVE_UNISTD_H
+
+/* debug build */
+#cmakedefine LIBHPDF_DEBUG
+
+/* debug trace enabled */
+#cmakedefine LIBHPDF_DEBUG_TRACE
+
+/* libpng is not available */
+#cmakedefine LIBHPDF_HAVE_NOPNGLIB
+
+/* zlib is not available */
+#cmakedefine LIBHPDF_HAVE_NOZLIB
+
+/* Define to the address where bug reports for this package should be sent. */
+#cmakedefine LIBHPDF_PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#cmakedefine LIBHPDF_PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#cmakedefine LIBHPDF_PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#cmakedefine LIBHPDF_PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#cmakedefine LIBHPDF_PACKAGE_VERSION
+
+/* Define to 1 if you have the ANSI C header files. */
+#cmakedefine LIBHPDF_STDC_HEADERS
+
+/* Define to `unsigned int' if <sys/types.h> does not define. */
+#cmakedefine LIBHPDF_size_t
diff -rupdN libharu-2.3.0RC2.orig/src/hpdf_image_ccitt.c libharu-2.3.0RC2/src/hpdf_image_ccitt.c
--- libharu-2.3.0RC2.orig/src/hpdf_image_ccitt.c	Fri Jan 14 16:00:31 2011
+++ libharu-2.3.0RC2/src/hpdf_image_ccitt.c	Fri Jan 14 16:01:00 2011
@@ -750,7 +750,7 @@ HPDF_Image_Load1BitImageFromMem  (HPDF_M
  *      TRUE if image is oriented TOP-BOTTOM;
  *      FALSE if image is oriented BOTTOM-TOP
  */
-HPDF_Image
+HPDF_EXPORT(HPDF_Image)
 HPDF_Image_LoadRaw1BitImageFromMem  (HPDF_Doc           pdf,
                            const HPDF_BYTE   *buf,
                           HPDF_UINT          width,
--- libharu-2.3.0RC2.orig/CMakeLists.txt	Fri Jan  7 15:27:41 2011
+++ libharu-2.3.0RC2/CMakeLists.txt	Sun Mar  4 14:59:13 2012
@@ -35,7 +35,7 @@
 cmake_minimum_required(VERSION 2.4.8 FATAL_ERROR)

 # Location where the haru cmake build system first looks for cmake modules
-set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake/modules)


 # set library name, msvc does not append 'lib' automatically
@@ -169,7 +169,6 @@ install(FILES ${haru_HDRS} DESTINATION i
 
 # install various files
 install(FILES README CHANGES INSTALL DESTINATION .)
-install(DIRECTORY doc DESTINATION .)
 if(NOT DEVPAK)
   install(DIRECTORY if DESTINATION .)
 endif(NOT DEVPAK)
