diff -rupd mariadb-native-client.orig/CMakeLists.txt mariadb-native-client/CMakeLists.txt
--- mariadb-native-client.orig/CMakeLists.txt	2012-11-29 04:29:20.000000000 +0100
+++ mariadb-native-client/CMakeLists.txt	2013-10-12 17:12:09.143269681 +0200
@@ -14,6 +14,8 @@ IF(COMMAND CMAKE_POLICY)
   CMAKE_POLICY(SET CMP0003 NEW)
 ENDIF(COMMAND CMAKE_POLICY)
 
+SET(DEBUG_LIB_POSTFIX "d" CACHE STRING "String appended to debug libraries")
+
 IF(WIN32)
   # Speedup system tests
   INCLUDE(${CMAKE_SOURCE_DIR}/WindowsCache.cmake)
@@ -111,12 +113,6 @@ IF(BUILD_DOCS)
 ENDIF()
 
 
-IF(WIN32)
- SET(CMAKE_INSTALL_PREFIX "")
-ENDIF()
-
-
-
 SET(CPACK_PACKAGE_VENDOR "Monty Program AB")
 SET(CPACK_PACKAGE_DESCRIPTION "MariaDB client library. A library for connecting to MariaDB and MySQL servers")
 SET(CPACK_PACKAGE_NAME "mariadb_client")
diff -rupd mariadb-native-client.orig/libmysql/CMakeLists.txt mariadb-native-client/libmysql/CMakeLists.txt
--- mariadb-native-client.orig/libmysql/CMakeLists.txt	2012-11-29 04:29:20.000000000 +0100
+++ mariadb-native-client/libmysql/CMakeLists.txt	2013-10-12 17:12:09.143269681 +0200
@@ -132,6 +132,9 @@ TARGET_LINK_LIBRARIES(mariadbclient ${CM
 IF(OPENSSL_FOUND)
   TARGET_LINK_LIBRARIES(mariadbclient ${OPENSSL_LIBRARIES})
 ENDIF()
+SET_TARGET_PROPERTIES(mariadbclient PROPERTIES
+   DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 ADD_LIBRARY(libmariadb SHARED ${LIBMYSQL_SOURCES})
 TARGET_LINK_LIBRARIES(libmariadb ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS})
@@ -156,9 +159,12 @@ ENDIF()
 
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION 
-   ${CPACK_PACKAGE_VERSION_MAJOR}
-   SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
+SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+   VERSION ${CPACK_PACKAGE_VERSION_MAJOR}
+   SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR}
+   PREFIX ""
+   DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 #
 # Installation
@@ -166,7 +172,7 @@ SET_TARGET_PROPERTIES(libmariadb PROPERT
 
 INSTALL(TARGETS
           libmariadb mariadbclient
-          RUNTIME DESTINATION "lib"
+          RUNTIME DESTINATION "bin"
           LIBRARY DESTINATION "lib"
           ARCHIVE DESTINATION "lib")
 
diff -rupd mariadb-native-client.orig/libmysql/my_stmt.c mariadb-native-client/libmysql/my_stmt.c
--- mariadb-native-client.orig/libmysql/my_stmt.c	2012-11-29 04:29:20.000000000 +0100
+++ mariadb-native-client/libmysql/my_stmt.c	2013-10-12 17:12:55.599271537 +0200
@@ -637,7 +637,7 @@ unsigned char* mysql_stmt_execute_genera
 
 mem_error:
   SET_CLIENT_STMT_ERROR(stmt, CR_OUT_OF_MEMORY, SQLSTATE_UNKNOWN, 0);
-  my_free((gptr)start, MYF(ALLOW_ZERO_PTR));
+  my_free((gptr)start, MYF(MY_ALLOW_ZERO_PTR));
   *request_len= 0;
   DBUG_RETURN(NULL);
 }
