diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd02de7..d86b157 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,6 +108,7 @@ IF(WIN32)
 ENDIF()
 
 IF(NOT IS_SUBPROJECT)
+SET(DEBUG_LIB_POSTFIX "d" CACHE STRING "String appended to debug libraries")
 IF(MSVC)
   # Speedup system tests
   INCLUDE(${CC_SOURCE_DIR}/cmake/WindowsCache.cmake)
diff --git a/cmake/CheckFunctions.cmake b/cmake/CheckFunctions.cmake
index d3d3be6..ea259dd 100644
--- a/cmake/CheckFunctions.cmake
+++ b/cmake/CheckFunctions.cmake
@@ -11,6 +11,8 @@
 # You will find the appropiate defines in 
 # include/my_config.h.in
 
+SET(CMAKE_REQUIRED_INCLUDES "stdlib.h")
+
 INCLUDE(CheckFunctionExists)
 
 CHECK_FUNCTION_EXISTS (access HAVE_ACCESS)
diff --git a/cmake/version_info.cmake b/cmake/version_info.cmake
index 4b2324a..36894ff 100644
--- a/cmake/version_info.cmake
+++ b/cmake/version_info.cmake
@@ -9,14 +9,14 @@ FUNCTION(GET_FILE_VERSION FILE_NAME FILE_VERSION)
 
   # if we build from a git repository, we calculate the file version:
   #  Patch number is numer of commits for given file
-  IF(EXISTS ${CC_SOURCE_DIR}/.git)
-    EXECUTE_PROCESS(COMMAND git --git-dir=${CC_SOURCE_DIR}/.git --work-tree=${CC_SOURCE_DIR} rev-list HEAD --count -- ${FILE_NAME} 
-      OUTPUT_VARIABLE FV)
-    STRING(REPLACE "\n" "" FV ${FV})
-    SET(${FILE_VERSION} ${FV} PARENT_SCOPE)
-  ELSE()
+#  IF(EXISTS ${CC_SOURCE_DIR}/.git)
+#    EXECUTE_PROCESS(COMMAND git --git-dir=${CC_SOURCE_DIR}/.git --work-tree=${CC_SOURCE_DIR} rev-list HEAD --count -- ${FILE_NAME} 
+#      OUTPUT_VARIABLE FV)
+#    STRING(REPLACE "\n" "" FV ${FV})
+#    SET(${FILE_VERSION} ${FV} PARENT_SCOPE)
+#  ELSE()
     SET(${FILE_VERSION} 0)
-  ENDIF()
+#  ENDIF()
 ENDFUNCTION()
 
 MACRO(SET_VERSION_INFO)
diff --git a/include/ma_pthread.h b/include/ma_pthread.h
index 7383367..f5d7ffa 100644
--- a/include/ma_pthread.h
+++ b/include/ma_pthread.h
@@ -28,6 +28,8 @@
 #define ETIME ETIMEDOUT				/* For FreeBSD */
 #endif
 
+#define _TIMESPEC_DEFINED
+
 #ifdef  __cplusplus
 extern "C" {
 #endif /* __cplusplus */ 
diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index d1472d1..0f015df 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -401,7 +401,10 @@ ENDIF()
 SET_TARGET_PROPERTIES(mariadbclient PROPERTIES IMPORTED_INTERFACE_LINK_LIBRARIES "${SYSTEM_LIBS}")
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES IMPORTED_INTERFACE_LINK_LIBRARIES "${SYSTEM_LIBS}")
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
+SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+  PREFIX ""
+  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION
  ${CPACK_PACKAGE_VERSION_MAJOR}
@@ -430,10 +433,14 @@ ENDIF()
 
 INSTALL(TARGETS mariadbclient
           COMPONENT Development
-          DESTINATION ${INSTALL_LIBDIR})
+          RUNTIME DESTINATION "bin"
+          LIBRARY DESTINATION "lib"
+          ARCHIVE DESTINATION "lib")
 INSTALL(TARGETS libmariadb
           COMPONENT SharedLibraries
-        DESTINATION ${INSTALL_LIBDIR})
+          RUNTIME DESTINATION "bin"
+          LIBRARY DESTINATION "lib"
+          ARCHIVE DESTINATION "lib")
 
 IF(WIN32)
    # On Windows, install PDB
