diff --git a/CMakeLists.txt b/CMakeLists.txt
index d800d98..4dee2ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,34 +29,35 @@ ELSE()
 ENDIF()
 
 OPTION(WITH_EXTERNAL_ZLIB "Enables use of external zlib" OFF)
+SET(DEBUG_LIB_POSTFIX "d" CACHE STRING "String appended to debug libraries")
 ###############
 
 IF(WITH_RTC)
   SET(RTC_OPTIONS "/RTC1 /RTCc")
 ENDIF()
 
-IF(MSVC)
-  # Speedup system tests
-  INCLUDE(${CMAKE_SOURCE_DIR}/cmake/WindowsCache.cmake)
-  IF (MSVC)
-    SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
-    FOREACH(BUILD_TYPE ${CONFIG_TYPES})
-      FOREACH(COMPILER CXX C)
-        SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
-        IF (NOT COMPILER_FLAGS STREQUAL "")
-          STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
-          IF (CMAKE_BUILD_TYPE STREQUAL "Debug")
-            SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
-            STRING(REPLACE "/Zi" "/ZI" COMPILER_FLAGS ${COMPILER_FLAGS})
-          ENDIF()
-          MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}") 
-          SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS} CACHE 
-               STRING "overwritten by libmariadb" FORCE)
-        ENDIF()
-      ENDFOREACH()
-    ENDFOREACH()
-  ENDIF()
-ENDIF()
+#IF(MSVC)
+#  # Speedup system tests
+#  INCLUDE(${CMAKE_SOURCE_DIR}/cmake/WindowsCache.cmake)
+#  IF (MSVC)
+#    SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
+#    FOREACH(BUILD_TYPE ${CONFIG_TYPES})
+#      FOREACH(COMPILER CXX C)
+#        SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
+#        IF (NOT COMPILER_FLAGS STREQUAL "")
+#          STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
+#          IF (CMAKE_BUILD_TYPE STREQUAL "Debug")
+#            SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
+#            STRING(REPLACE "/Zi" "/ZI" COMPILER_FLAGS ${COMPILER_FLAGS})
+#          ENDIF()
+#          MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}") 
+#          SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS} CACHE 
+#               STRING "overwritten by libmariadb" FORCE)
+#        ENDIF()
+#      ENDFOREACH()
+#    ENDFOREACH()
+#  ENDIF()
+#ENDIF()
 
 # Disable dbug information for release builds 
 SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -DDBUG_OFF")
@@ -214,9 +215,9 @@ IF(CLIENT_DOCS)
           DESTINATION ${DOCS_INSTALL_DIR_${INSTALL_LAYOUT}})
 ENDIF()
 
-IF(MSVC)
- SET(CMAKE_INSTALL_PREFIX "")
-ENDIF()
+#IF(MSVC)
+# SET(CMAKE_INSTALL_PREFIX "")
+#ENDIF()
 
 IF(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
   ADD_SUBDIRECTORY(win/packaging)
diff --git a/cmake/CheckFunctions.cmake b/cmake/CheckFunctions.cmake
index 8a2208f..69d2b74 100644
--- a/cmake/CheckFunctions.cmake
+++ b/cmake/CheckFunctions.cmake
@@ -3,6 +3,8 @@
 # include/my_config.h.in
 # You will find the appropiate defines in 
 
+SET(CMAKE_REQUIRED_INCLUDES "stdlib.h")
+
 INCLUDE(CheckFunctionExists)
 
 CHECK_FUNCTION_EXISTS (access HAVE_ACCESS)
diff --git a/include/my_pthread.h b/include/my_pthread.h
index 2117868..043e98b 100644
--- a/include/my_pthread.h
+++ b/include/my_pthread.h
@@ -25,6 +25,8 @@
 #define ETIME ETIMEDOUT				/* For FreeBSD */
 #endif
 
+#define _TIMESPEC_DEFINED
+
 #ifdef  __cplusplus
 extern "C" {
 #endif /* __cplusplus */ 
diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index 3ea50db..45acfa8 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -378,7 +378,10 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
   TARGET_LINK_LIBRARIES (mariadbclient "-Wl,--version-script=${EXPORT_FILE}")
 ENDIF()
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
+SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+  PREFIX ""
+  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION 
    ${CPACK_PACKAGE_VERSION_MAJOR}
@@ -419,9 +422,9 @@ ENDIF()
 
 INSTALL(TARGETS
           libmariadb mariadbclient
-          RUNTIME DESTINATION "${LIB_INSTALL_DIR}/${SUFFIX_INSTALL_DIR}"
-          LIBRARY DESTINATION "${LIB_INSTALL_DIR}/${SUFFIX_INSTALL_DIR}"
-          ARCHIVE DESTINATION "${LIB_INSTALL_DIR}/${SUFFIX_INSTALL_DIR}")
+          RUNTIME DESTINATION "bin"
+          LIBRARY DESTINATION "lib"
+          ARCHIVE DESTINATION "lib")
 
 IF(NOT WIN32 AND WITH_MYSQLCOMPAT)
   INSTALL(FILES "${CMAKE_BINARY_DIR}/libmariadb/libmysql${CMAKE_SHARED_LIBRARY_SUFFIX}"
