diff -ru mariadb-src/CMakeLists.txt mariadb-src-modified/CMakeLists.txt
--- mariadb-src/CMakeLists.txt	2014-04-04 12:21:01.000000000 +0200
+++ mariadb-src-modified/CMakeLists.txt	2014-04-23 15:42:48.165780100 +0200
@@ -27,34 +27,35 @@
 
 OPTION(WITH_SQLITE "Enables Sqlite support" OFF)
 OPTION(WITH_EXTERNAL_ZLIB "Enables use of external zlib" OFF)
+SET(DEBUG_LIB_POSTFIX "d" CACHE STRING "String appended to debug libraries")
 ###############
 
 IF(WITH_RTC)
   SET(RTC_OPTIONS "/RTC1 /RTCc")
 ENDIF()
 
-IF(MSVC)
-  # Speedup system tests
-  INCLUDE(${CMAKE_SOURCE_DIR}/cmake/WindowsCache.cmake)
-  IF (MSVC)
-    SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
-    FOREACH(BUILD_TYPE ${CONFIG_TYPES})
-      FOREACH(COMPILER CXX C)
-        SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
-        IF (NOT COMPILER_FLAGS STREQUAL "")
-          STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
-          IF (CMAKE_BUILD_TYPE STREQUAL "Debug")
-            SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
-            STRING(REPLACE "/Zi" "/ZI" COMPILER_FLAGS ${COMPILER_FLAGS})
-          ENDIF()
-          MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}") 
-          SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS} CACHE 
-               STRING "overwritten by libmariadb" FORCE)
-        ENDIF()
-      ENDFOREACH()
-    ENDFOREACH()
-  ENDIF()
-ENDIF()
+#IF(MSVC)
+#  # Speedup system tests
+#  INCLUDE(${CMAKE_SOURCE_DIR}/cmake/WindowsCache.cmake)
+#  IF (MSVC)
+#    SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
+#    FOREACH(BUILD_TYPE ${CONFIG_TYPES})
+#      FOREACH(COMPILER CXX C)
+#        SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
+#        IF (NOT COMPILER_FLAGS STREQUAL "")
+#          STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
+#          IF (CMAKE_BUILD_TYPE STREQUAL "Debug")
+#            SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
+#            STRING(REPLACE "/Zi" "/ZI" COMPILER_FLAGS ${COMPILER_FLAGS})
+#          ENDIF()
+#          MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}") 
+#          SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS} CACHE 
+#               STRING "overwritten by libmariadb" FORCE)
+#        ENDIF()
+#      ENDFOREACH()
+#    ENDFOREACH()
+#  ENDIF()
+#ENDIF()
 
 # Disable dbug information for release builds 
 SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -DDBUG_OFF")
diff -ru mariadb-src-modified/CMakeLists.txt mariadb-src/CMakeLists.txt
--- mariadb-src-modified/CMakeLists.txt 2014-04-23 15:42:48.165780100 +0200
+++ mariadb-src/CMakeLists.txt  2014-04-29 09:58:51.532821500 +0200
@@ -191,9 +191,9 @@
 ENDIF()


-IF(MSVC)
- SET(CMAKE_INSTALL_PREFIX "")
-ENDIF()
+#IF(MSVC)
+# SET(CMAKE_INSTALL_PREFIX "")
+#ENDIF()

 IF(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
   ADD_SUBDIRECTORY(win/packaging)
diff -ru mariadb-src/libmariadb/CMakeLists.txt mariadb-src-modified/libmariadb/CMakeLists.txt
--- mariadb-src/libmariadb/CMakeLists.txt	2014-04-04 12:21:01.000000000 +0200
+++ mariadb-src-modified/libmariadb/CMakeLists.txt	2014-04-23 15:43:46.712128700 +0200
@@ -169,7 +169,10 @@
   TARGET_LINK_LIBRARIES (libmariadb "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/version_script.txt")
 ENDIF()
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
+SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+  PREFIX ""
+  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION 
    ${CPACK_PACKAGE_VERSION_MAJOR}
@@ -210,7 +213,7 @@
 
 INSTALL(TARGETS
           libmariadb mariadbclient
-          RUNTIME DESTINATION "lib/mariadb"
+          RUNTIME DESTINATION "bin/mariadb"
           LIBRARY DESTINATION "lib/mariadb"
           ARCHIVE DESTINATION "lib/mariadb")
 IF(NOT WIN32 AND WITH_MYSQLCOMPAT)
