diff --color -rupd mariadb-connector-c-orig/cmake/CheckFunctions.cmake mariadb-connector-c/cmake/CheckFunctions.cmake
--- mariadb-connector-c-orig/cmake/CheckFunctions.cmake	2021-05-03 13:58:17.000000000 +0200
+++ mariadb-connector-c/cmake/CheckFunctions.cmake	2021-06-24 16:24:30.937853403 +0200
@@ -11,6 +11,8 @@
 # You will find the appropriate defines in 
 # include/my_config.h.in
 
+set(CMAKE_REQUIRED_INCLUDES "stdlib.h")
+
 INCLUDE(CheckFunctionExists)
 
 CHECK_FUNCTION_EXISTS (alloca HAVE_ALLOCA)
@@ -27,4 +29,4 @@ IF(HAVE_FILE_UCONTEXT_H)
   CHECK_FUNCTION_EXISTS (makecontext HAVE_UCONTEXT_H)
 ENDIF()
 
-CHECK_FUNCTION_EXISTS (cuserid HAVE_CUSERID)
\ No newline at end of file
+CHECK_FUNCTION_EXISTS (cuserid HAVE_CUSERID)
diff --color -rupd mariadb-connector-c-orig/cmake/version_info.cmake mariadb-connector-c/cmake/version_info.cmake
--- mariadb-connector-c-orig/cmake/version_info.cmake	2021-05-03 13:58:17.000000000 +0200
+++ mariadb-connector-c/cmake/version_info.cmake	2021-06-24 16:24:51.213816945 +0200
@@ -9,14 +9,14 @@ FUNCTION(GET_FILE_VERSION FILE_NAME FILE
 
   # if we build from a git repository, we calculate the file version:
   #  Patch number is number of commits for given file
-  IF(GIT_EXECUTABLE AND EXISTS ${CC_SOURCE_DIR}/.git)
-    EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} --git-dir=${CC_SOURCE_DIR}/.git --work-tree=${CC_SOURCE_DIR} rev-list HEAD --count -- ${FILE_NAME} 
-      OUTPUT_VARIABLE FV)
-    STRING(REPLACE "\n" "" FV ${FV})
-    SET(${FILE_VERSION} ${FV} PARENT_SCOPE)
-  ELSE()
+  #IF(GIT_EXECUTABLE AND EXISTS ${CC_SOURCE_DIR}/.git)
+  #  EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} --git-dir=${CC_SOURCE_DIR}/.git --work-tree=${CC_SOURCE_DIR} rev-list HEAD --count -- ${FILE_NAME} 
+  #    OUTPUT_VARIABLE FV)
+  #  STRING(REPLACE "\n" "" FV ${FV})
+  #  SET(${FILE_VERSION} ${FV} PARENT_SCOPE)
+  #ELSE()
     SET(${FILE_VERSION} 0)
-  ENDIF()
+  #ENDIF()
 ENDFUNCTION()
 
 MACRO(SET_VERSION_INFO)
diff --color -rupd mariadb-connector-c-orig/CMakeLists.txt mariadb-connector-c/CMakeLists.txt
--- mariadb-connector-c-orig/CMakeLists.txt	2021-05-03 13:58:17.000000000 +0200
+++ mariadb-connector-c/CMakeLists.txt	2021-06-24 16:25:44.741720584 +0200
@@ -93,30 +93,32 @@ IF(WIN32)
   INCLUDE(${CC_SOURCE_DIR}/cmake/version_info.cmake)
 ENDIF()
 
+SET(DEBUG_LIB_POSTFIX "d" CACHE STRING "String appended to debug libraries")
+
 IF(NOT IS_SUBPROJECT)
 IF(MSVC)
   # Speedup system tests
-  INCLUDE(${CC_SOURCE_DIR}/cmake/WindowsCache.cmake)
+  #INCLUDE(${CC_SOURCE_DIR}/cmake/WindowsCache.cmake)
   ADD_DEFINITIONS(-DWIN32_LEAN_AND_MEAN)
-  IF (MSVC)
-    SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
-    FOREACH(BUILD_TYPE ${CONFIG_TYPES})
-      FOREACH(COMPILER CXX C)
-        SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
-        IF (NOT COMPILER_FLAGS STREQUAL "")
-          IF(NOT WITH_ASAN)
-            STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
-            IF (BUILD_TYPE STREQUAL "DEBUG")
-              SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
-            ENDIF()
-          ENDIF()
-          STRING(REPLACE "/Zi" "/Z7" COMPILER_FLAGS ${COMPILER_FLAGS})
-          MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}")
-          SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS})
-        ENDIF()
-      ENDFOREACH()
-    ENDFOREACH()
-  ENDIF()
+  #IF (MSVC)
+  #  SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
+  #  FOREACH(BUILD_TYPE ${CONFIG_TYPES})
+  #    FOREACH(COMPILER CXX C)
+  #      SET(COMPILER_FLAGS "${CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}}")
+  #      IF (NOT COMPILER_FLAGS STREQUAL "")
+  #        IF(NOT WITH_ASAN)
+  #          STRING(REPLACE "/MD" "/MT" COMPILER_FLAGS ${COMPILER_FLAGS})
+  #          IF (BUILD_TYPE STREQUAL "DEBUG")
+  #            SET(COMPILER_FLAGS "${COMPILER_FLAGS} ${RTC_OPTIONS}")
+  #          ENDIF()
+  #        ENDIF()
+  #        STRING(REPLACE "/Zi" "/Z7" COMPILER_FLAGS ${COMPILER_FLAGS})
+  #        MESSAGE (STATUS "CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE}= ${COMPILER_FLAGS}")
+  #        SET(CMAKE_${COMPILER}_FLAGS_${BUILD_TYPE} ${COMPILER_FLAGS})
+  #      ENDIF()
+  #    ENDFOREACH()
+  #  ENDFOREACH()
+  #ENDIF()
 ENDIF()
 ENDIF(NOT IS_SUBPROJECT)
 
diff --color -rupd mariadb-connector-c-orig/libmariadb/CMakeLists.txt mariadb-connector-c/libmariadb/CMakeLists.txt
--- mariadb-connector-c-orig/libmariadb/CMakeLists.txt	2021-05-03 13:58:17.000000000 +0200
+++ mariadb-connector-c/libmariadb/CMakeLists.txt	2021-06-24 16:27:05.653574556 +0200
@@ -426,7 +426,15 @@ ENDIF()
 SET_TARGET_PROPERTIES(mariadbclient PROPERTIES IMPORTED_INTERFACE_LINK_LIBRARIES "${SYSTEM_LIBS}")
 SET_TARGET_PROPERTIES(libmariadb PROPERTIES IMPORTED_INTERFACE_LINK_LIBRARIES "${SYSTEM_LIBS}")
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
+SET_TARGET_PROPERTIES(mariadbclient PROPERTIES
+  PREFIX ""
+  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
+
+SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+  PREFIX ""
+  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+)
 
 #
 # Installation
