--- openssl-orig/build.info	2019-06-17 13:15:15.107088156 +0200
+++ openssl/build.info	2019-06-14 16:05:20.352948218 +0200
@@ -86,8 +86,13 @@ ELSIF[{- $config{target} =~ /^mingw/ -}]
  SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
  SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
 ELSIF[{- $config{target} =~ /^VC-/ -}]
- SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $target{multilib} -}
- SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $target{multilib} -}
+ IF[{- $config{build_type} eq "debug" -}]
+  SHARED_NAME[libcrypto]=libcryptod
+  SHARED_NAME[libssl]=libssld
+ ELSE
+  SHARED_NAME[libcrypto]=libcrypto
+  SHARED_NAME[libssl]=libssl
+ ENDIF
 ENDIF
 
 # VMS has a cultural standard where all libraries are prefixed.
