--- orig/postgresql-9.3.5/src/interfaces/libpq/win32.mak	2014-07-21 21:10:42.000000000 +0200
+++ postgresql-9.3.5/src/interfaces/libpq/win32.mak	2014-09-23 22:16:18.864503000 +0200
@@ -78,7 +78,7 @@
 !ENDIF


-ALL : config "$(OUTDIR)\$(OUTFILENAME).lib" "$(OUTDIR)\$(OUTFILENAME).dll"
+ALL : config "$(OUTDIR)\$(OUTFILENAME).lib" #"$(OUTDIR)\$(OUTFILENAME).dll"

 CLEAN :
        -@erase "$(INTDIR)\getaddrinfo.obj"
@@ -115,6 +115,7 @@
 	-@erase "$(INTDIR)\open.obj"
 	-@erase "$(INTDIR)\win32error.obj"
 	-@erase "$(INTDIR)\win32setlocale.obj"
+	-@erase "$(INTDIR)\fe_memutils.obj"
 	-@erase "$(OUTDIR)\$(OUTFILENAME).lib"
 	-@erase "$(OUTDIR)\$(OUTFILENAME)dll.lib"
 	-@erase "$(OUTDIR)\libpq.res"
@@ -161,7 +162,8 @@
 	"$(INTDIR)\open.obj" \
 	"$(INTDIR)\win32error.obj" \
 	"$(INTDIR)\win32setlocale.obj" \
-	"$(INTDIR)\pthread-win32.obj"
+	"$(INTDIR)\pthread-win32.obj" \
+	"$(INTDIR)\fe_memutils.obj"
 
 
 config: ..\..\include\pg_config.h ..\..\include\pg_config_ext.h pg_config_paths.h  ..\..\include\pg_config_os.h
@@ -208,7 +210,7 @@
 LINK32=link.exe
 LINK32_FLAGS=kernel32.lib user32.lib advapi32.lib shfolder.lib wsock32.lib ws2_32.lib secur32.lib $(SSL_LIBS)  $(KFW_LIB) $(ADD_SECLIB) \
  /nologo /subsystem:windows /dll $(LOPT) /incremental:no \
- /pdb:"$(OUTDIR)\libpqdll.pdb" /machine:$(CPU) \
+ /pdb:"$(OUTDIR)\libpqdll.pdb" \
  /out:"$(OUTDIR)\$(OUTFILENAME).dll"\
  /implib:"$(OUTDIR)\$(OUTFILENAME)dll.lib"  \
  /libpath:"$(SSL_LIB_PATH)" /libpath:"$(KFW_LIB_PATH)" \
@@ -235,9 +237,11 @@
 # Inclusion of manifest
 !IF "$(_NMAKE_VER)" != "6.00.8168.0" && "$(_NMAKE_VER)" != "7.00.9466"
 !IF "$(_NMAKE_VER)" != "6.00.9782.0" && "$(_NMAKE_VER)" != "7.10.3077"
+!IF EXIST ($(OUTDIR)\$(OUTFILENAME).dll.manifest)
         mt -manifest $(OUTDIR)\$(OUTFILENAME).dll.manifest -outputresource:$(OUTDIR)\$(OUTFILENAME).dll;2
 !ENDIF
 !ENDIF
+!ENDIF
 
 "$(INTDIR)\getaddrinfo.obj" : ..\..\port\getaddrinfo.c
 	$(CPP) @<<
@@ -345,6 +349,11 @@
 	$(CPP_PROJ) /I"." ..\..\port\win32setlocale.c
 <<
 
+"$(INTDIR)\fe_memutils.obj" : ..\..\common\fe_memutils.c
+	$(CPP) @<<
+	$(CPP_PROJ) /I"." ..\..\common\fe_memutils.c
+<<
+
 .c{$(CPP_OBJS)}.obj:
 	$(CPP) $(CPP_PROJ) $<
 
