diff -rupNd gd.orig/CMakeLists.txt gd/CMakeLists.txt
--- gd.orig/CMakeLists.txt	Tue Nov 27 09:30:34 2007
+++ gd/CMakeLists.txt	Fri Jan 14 04:11:31 2011
@@ -169,11 +169,16 @@ ENDIF (NOT WIN32)
 
 	add_library(${GD_LIB} ${LIBGD_SRC_FILES})
 
-	if (WIN32)
-		SET_TARGET_PROPERTIES(${GD_LIB} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:msvcrt.lib")
-	ENDIF(WIN32)
+#	if (WIN32)
+#		SET_TARGET_PROPERTIES(${GD_LIB} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:msvcrt.lib")
+#	ENDIF(WIN32)
 
 	target_link_libraries(${GD_LIB} ${FREETYPE_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} ${XPM_LIBRARIES} ${FONTCONFIG_LIBRARIES})
+
+    install(TARGETS ${GD_LIB}
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib
+        LIBRARY DESTINATION lib )
 
 endif (USE_EXT_GD)
 
diff -rupNd gd.orig/cmake/modules/FindPNG.cmake gd/cmake/modules/FindPNG.cmake
--- gd.orig/cmake/modules/FindPNG.cmake	Tue Nov 27 09:30:34 2007
+++ gd/cmake/modules/FindPNG.cmake	Thu Jan  1 01:00:00 1970
@@ -1,63 +0,0 @@
-# - Find the native PNG includes and library
-#
-
-# This module defines
-#  PNG_INCLUDE_DIR, where to find png.h, etc.
-#  PNG_LIBRARIES, the libraries to link against to use PNG.
-#  PNG_DEFINITIONS - You should ADD_DEFINITONS(${PNG_DEFINITIONS}) before compiling code that includes png library files.
-#  PNG_FOUND, If false, do not try to use PNG.
-# also defined, but not for general use are
-#  PNG_LIBRARY, where to find the PNG library.
-# None of the above will be defined unles zlib can be found.
-# PNG depends on Zlib
-#
-# Copyright (c) 2002 Kitware, Inc., Insight Consortium.  All rights reserved.
-# See Copyright.txt or http://www.cmake.org/HTML/Copyright.html for details.
-
-
-INCLUDE(FindZLIB)
-
-SET(PNG_FOUND "NO")
-
-IF(ZLIB_FOUND)
-  FIND_PATH(PNG_PNG_INCLUDE_DIR png.h
-  /usr/local/include
-  /usr/include
-  /usr/local/include/libpng             # OpenBSD
-  )
-
-  SET(PNG_NAMES ${PNG_NAMES} png libpng)
-  FIND_LIBRARY(PNG_LIBRARY
-    NAMES ${PNG_NAMES}
-    PATHS /usr/lib64 /usr/lib /usr/local/lib
-  )
-
-  IF (PNG_LIBRARY AND PNG_PNG_INCLUDE_DIR)
-      # png.h includes zlib.h. Sigh.
-      SET(PNG_INCLUDE_DIR ${PNG_PNG_INCLUDE_DIR} ${ZLIB_INCLUDE_DIR} )
-      SET(PNG_LIBRARIES ${PNG_LIBRARY} ${ZLIB_LIBRARY})
-      SET(PNG_FOUND "YES")
-			SET(HAVE_PNG_H)
-      IF (CYGWIN)
-        IF(BUILD_SHARED_LIBS)
-           # No need to define PNG_USE_DLL here, because it's default for Cygwin.
-        ELSE(BUILD_SHARED_LIBS)
-          SET (PNG_DEFINITIONS -DPNG_STATIC)
-        ENDIF(BUILD_SHARED_LIBS)
-      ENDIF (CYGWIN)
-
-  ENDIF (PNG_LIBRARY AND PNG_PNG_INCLUDE_DIR)
-
-ENDIF(ZLIB_FOUND)
-
-IF (PNG_FOUND)
-  IF (NOT PNG_FIND_QUIETLY)
-    MESSAGE(STATUS "Found PNG: ${PNG_LIBRARY}")
-  ENDIF (NOT PNG_FIND_QUIETLY)
-ELSE (PNG_FOUND)
-  IF (PNG_FIND_REQUIRED)
-    MESSAGE(FATAL_ERROR "Could not find PNG library")
-  ENDIF (PNG_FIND_REQUIRED)
-ENDIF (PNG_FOUND)
-
-MARK_AS_ADVANCED(PNG_PNG_INCLUDE_DIR PNG_LIBRARY )
